main.go

```diff
@@ -103,14 +105,25 @@ func main() {
 				Source: ccc.TokenSource(ctx),
 			},
 		}
-
 	}
 
-	u, err := url.Parse(cfg.observatoriumURL)
-	if err != nil {
-		log.Fatalf("failed to parse Observatorium API URL: %v", err)
+	var u *url.URL
+
+	if cfg.rulesBackendURL != "" {
+		u, err := url.Parse(cfg.rulesBackendURL)
```

Comment:
when this is the case, we should really use the OpenAPI-generated client. There is no guarantee that the rules backend API and the Observatorium API will have the same interface, so we should adapt the `getRules` call appropriately

Reply:
I have refactored the code to use a `fetcher` interface for all the rule fetching related logic.